### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by alicemirigo92**
- [ ] Use ruby gems as software packages system.
- [ ] Install Ruby on Rails framework.
- [ ] Understand Rails RESTful design and router.
- [ ] Use controllers to handle requests and render empty views.
- [ ] Use params from browser request in a safe way.
- [ ] Use preprocessed html file with embedded Ruby code.
- [ ] Use layouts and templates for shared content.
- [ ] Use database migration files to maintain database schema.
- [ ] Use validations for models.
- [ ] Secure app from n+1 problems.
- [ ] Understand what ORM is.
- [ ] Write SQL queries with ActiveRecord.
- [ ] Set up associations between models.
- [ ] Build a webapp that requires the user to log in.
- [ ] Use devise gem for authentication.
- [ ] Limit access to webapp resources based on authorization rules.
- [ ] Analyze in writing why you have made a coding choice using one structure over another.

### CODE REVIEW

File: app/views/devise/sessions/splash.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="top p-2">
+    <h2 class='text-light text-center h1'>Welcome</h2>
+</div>
+
+<div class="splash">
+    <div class='d-flex splash-footer p-3 justify-content-between gap-2'>
+      <%= link_to 'Sign In', new_user_session_path, class: 'btn border border-light p-2 text-light col-6' %>
+      <%= link_to 'Sign Up', new_user_registration_path, class: 'btn border border-light text-light col-6 p-2' %>
+    </div>
+</div>
```
Comment (MahmoudBakr23) [inline app/views/devise/sessions/splash.html.erb]: - [ ] For the splash screen, please make sure to follow the [design guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=You%20should%20follow%20these%20design%20guidelines%2C%20including%3A) for this page and other pages of the app. Also, please make sure to display the name of the app in the splash screen as required. Please take a look at the screens below.

Screenshot (1)

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          <%= category.name %>
+        <% end %>
+        <p><%= category.created_at.strftime("%d %h %Y")  %></p>
+      </div>
+      <p></p>
+      <% @total = 0 %>
+      <% category.deal_categories.each { |e| @total += e.deal.amount }%>
+      <p class="text-end col-3">Total: $<%= @total %></p>
+    </div>
+  <% end %>
```
Comment (MahmoudBakr23) [inline app/views/categories/index.html.erb]: - [ ] Please make sure to adjust the design for the `categories` and `deals` index pages as in the design guidelines. Please see the link in the comment above.
Screenshot (1)

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @alicemirigo92 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the models, controllers, and views correctly
- Great work with the database and ERD
- Well done with the unit and integration tests
- Kudos for solving the n+1 problems
- Descriptive pr and readme
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long work hours are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/splash.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<div class="top p-2">
+    <h2 class='text-light text-center h1'>Welcome</h2>
+</div>
+
+<div class="splash">
+    <div class='d-flex splash-footer p-3 justify-content-around gap-2'>
+      <%= link_to 'Sign In', new_user_session_path, class: 'btn border border-light p-2 text-light col-6' %>
+      <%= link_to 'Sign Up', new_user_registration_path, class: 'btn border border-light text-light col-6 p-2' %>
+    </div>
+</div>
```
Comment (ishimwezachee) [inline app/views/devise/sessions/splash.html.erb]: - [ ]  As previous suggested please provide the name of your app on splash page and it should be in the center of the page as it is designed in the app. 

image

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @alicemirigo92 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
image

 ## Highlights

- Well done on creating the models, controllers, and views correctly
- Great work with the database and ERD
- Well done with the unit and integration tests
- Kudos for solving the n+1 problems
- Descriptive pr and readme
- Correct Linters


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @ishimwezachee, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice project implementation 🔥 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**